src/main/java/httpserver/HttpRouter.java

```diff
@@ -0,0 +1,82 @@
+package httpserver;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * An HttpRouter is used to route incoming requests to specific handlers.
+ *
+ * @see HttpHandler
+ * @see HttpRequest
+ */
+public class HttpRouter {
```

Comment:
This is a thought I had mainly because this class is now basically useless since for the most part future users should just be using `HttpServer` to do basically everything (see [`HandlerTest#serverSetup`](https://github.com/storytellersoftware/java-httpserver/blob/4539f034d529f159b08056da4b412d0746112d91/src/test/java/tests/HandlerTest.java#L26-L68) for an example).